applications/matter_bridge/src/ble_providers/ble_bridged_device_factory.cpp

```diff
@@ -334,10 +335,14 @@ CHIP_ERROR BleBridgedDeviceFactory::CreateDevice(int deviceType, bt_addr_le_t bt
 	BLEConnectivityManager::Instance().Recover(provider);
 	err = AddMatterDevices(reinterpret_cast<MatterBridgedDevice::DeviceType *>(&deviceType), 1, nodeLabel, provider,
 			       &index, &endpointId);
-exit:
 
 	if (err != CHIP_NO_ERROR) {
 		BLEConnectivityManager::Instance().RemoveBLEProvider(btAddress);
+		return err;
```

Comment:
Previously, we call also delete on the provider's memory if there was an error, now it returns failure but we don't call delete anymore. Is it ok?

Reply:
Yes, the object will be deleted in the `AddMatterDevices()`. The fact that we deleted the provider's memory after the aforementioned method returned an error was the direct reason for one of the observed crashes.